jekyll/_cci2/automate-llm-evaluation-testing-with-the-circleci-evals-orb.adoc

```diff
@@ -0,0 +1,248 @@
+
+With the CircleCI Evals orb, you can remove the manual work involved in triggering LLM evaluations, reviewing evaluation results, and determining whether the results meet set expectations.
+
+TIP: To learn more about LLM evaluations, read our article on xref:testing-llm-enabled-applications-through-evaluations#[methods for testing LLM-enabled applications through evaluations].
+
+== Concepts
+* **LLM Evaluations**: A method to assess the performance and accuracy of AI models and algorithms. Evaluations measure metrics such as task fidelity, consistency, relevance and coherence, or tone and style.
+
+* **Evaluation results**: The set of quantitative (often numeric) results output by running an evaluation.
+
+* **Metrics**: 1 or more units of data, used as inputs for testing.
+
+* **Evaluation test**: A test that determines if input metrics meet specified conditions.
+
+* **Test suite**: A set of test cases that are related and executed. Is a `.json` file that is assigned a name by the user.
+
+* **Test case**: A test scenario, tests a single concept to make it clear what has failed when the test does not pass. Each has:
+** *Name*: The name assigned to the test case. Usually descriptive of the scenario.
+** *Assertion*: A statement that checks if the specified condition is true. If false, the test fails.
+** *Result*: Each test case has a result of pass or fail.
+
+* **Evaluation test result**:
+** *Failure* indicates a proposed change resulted in a degradation of model performance. The job stops running, and the pipeline fails.
+** *Success* indicates that model performance has met set criteria, the job continues to run.
+
+image::eval-test-diagram-concepts.png[Evaluation testing: Concepts diagram]
+
+== Prerequisites
+
+* A CircleCI account connected to your code. You can sign up xref:first-steps#[for free].
+* A CircleCI project with a workflow configured to build your code.
+* CircleCI xref:contexts#[contexts] relevant for your workflow, which includes tools such as pre-trained model providers like OpenAI.
+* Existing LLM evaluations on a open source library or third-party tools such as Braintrust or Langsmith.
+
+== 1. Add the CircleCI Evals orb
+
+Use CircleCI version 2.1 at the top of your `.circleci/config.yml` file.
+
+[source ,yaml]
+----
+version: 2.1
+----
+
+Add the orbs stanza below your version, invoking the orb:
+
+[source ,yaml]
+----
+orbs:
+  evals: circleci/evals@2.0.0
+----
+
+== 2. Add and define an evaluation job
+
+Define a new job to run your LLM evaluations, save evaluation results and test them. It is important to include the step to save your evaluation results in order to run evaluation testing.
+
+[source ,yaml]
+----
+jobs:
+  eval-test:
+    docker: 
+      - image: cimg/python:stable
+    steps:
+      # Add an evaluation step
+      # - run: <your eval step> # Run evals and output results for the evaluation test
```

Comment:
👀 commented out eval run...I suppose this is okay to keep here.

Reply:
I commented it out since the user needs to define how their evals will run entirely (so this is very much a placeholder), compared to the evals/test which we provide an explicit framework around, does that makes sense? Or should it be left in, uncommented? 🤔

Reply:
If we leave that line commented-out, my concern is that some users might perceive it as optional. I think leaving it here uncommented would work better because your usage of `<your eval step>` is already a good placeholder indicator, much like the other `<>`-wrapped placeholders in the example config.

```suggestion
      - run: <your eval step> # Run evals and output results for the evaluation test
```

I am also open to others' opinions on this.

Reply:
I feel the same as ryan, comments are easy to miss. In an ideal world, the examples are functional and you can copy/paste them for quick starting. So I'd say it's more clear to have it uncommented.